### ISSUE TITLE
Stream not playing in Chrome / Firefox in Windows

### ISSUE THREAD

**Issue by avelad** [google/shaka-player#3144]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?** Yes

**What version of Shaka Player are you using?** 3.0.7

**Can you reproduce the issue with our latest release version?** Yes

**Can you reproduce the issue with the latest code from `master`?** Yes

**Are you using the demo app or your own custom app?** Both

**If custom app, can you reproduce the issue using our demo app?** Yes

**What browser and OS are you using?**
Firefox 85 - Windows 10
Chrome 88 - Windows 10

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**

**What are the manifest and license server URIs?**

https://streamer.zapitv.com/provision/mirage.mp4/index.mpd

**What did you do?**

Load the previous stream

**What did you expect to happen?**
The stream plays without problem

**What actually happened?**

The stream does not start playing and in the browser console I see this:
`Cannot play media. No decoders for requested formats: audio/mp4;codecs="mp4a.40.2";bitrate="122000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3676000";width="1920";height="1080"`

Note: The same stream in macOS works without problem.

**Comment by ziddey** [comment]
Has the stream changed since this was posted? Just tested ff85/chrome88 on win10 and it appears to play on both.

Do see in firefox:

```Cannot play media. No decoders for requested formats: audio/mp4;codecs="mp4a.40.2";bitrate="122000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3676000";width="1920";height="1080", audio/mp4;codecs="mp4a.40.2";bitrate="122000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3676000";width="1920";height="1080", audio/mp4;codecs="mp4a.40.2";bitrate="122000";channels="2", audio/mp4; codecs="mp4a.40.2", audio/mp4; codecs="mp4a.40.2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3676000";width="1920";height="1080", video/mp4; codecs="avc1.640028", video/mp4; codecs="avc1.640028", audio/mp4;codecs="mp4a.40.2";bitrate="122000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3676000";width="1920";height="1080"```

but it doesn't affect playback. For what it's worth, I see similar errors in firefox for pretty much any stream.

**Comment by avelad** [comment]
The problem is that it doesn't happen on all computers, on some it works and on others it doesn't.

**Comment by ziddey** [comment]
Ah, does it make a difference if hardware acceleration is disabled in chrome/ff?

**Comment by TheModMaker** [comment]
First, the URL you posted is no longer valid (404).  Secondly, that looks like normal H.264+AAC content, so I don't know why it wouldn't be supported.  It looks like it is multiplexed, but there are multiple MIME strings combined instead of combining their codec values.  Is that console line coming from our code or from the platform (i.e. does DevTools tell you where it comes from)?  Can you create non-multiplexed content for that stream and does that play?

**Comment by avelad** [comment]
The content provider has removed it, I will reopen the ticket when I get the stream again (it could take a couple of days ...)

**Comment by avelad** [comment]
@TheModMaker New stream:

https://shaka-player-demo.appspot.com/demo/#audiolang=es-419;textlang=es-419;jumpLargeGaps;uilang=es-419;asset=https://streamer.zapitv.com/provision/NO_GRITES.mp4/index.mpd;panel=HOME;build=uncompiled

"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:85.0) Gecko/20100101 Firefox/85.0"
![image](https://user-images.githubusercontent.com/8983024/109203998-1de83b00-77a5-11eb-8209-14631a3188f8.png)

`No se puede reproducir el medio. No hay decodificadores para los formatos solicitados: audio/mp4;codecs="mp4a.40.2";bitrate="124000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3363000";width="1920";height="1080", audio/mp4;codecs="mp4a.40.2";bitrate="124000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3363000";width="1920";height="1080", audio/mp4;codecs="mp4a.40.2";bitrate="124000";channels="2", audio/mp4; codecs="mp4a.40.2", audio/mp4; codecs="mp4a.40.2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3363000";width="1920";height="1080", video/mp4; codecs="avc1.640028", video/mp4; codecs="avc1.640028", audio/mp4;codecs="mp4a.40.2";bitrate="124000";channels="2", video/mp4;codecs="avc1.640028";framerate="25";bitrate="3363000";width="1920";height="1080"`

**Comment by avelad** [comment]
Note: the same stream in macOS works correctly.

**Comment by avelad** [comment]
I have found something very curious ... without the jumpLargeGaps setting it works. But if I activate it, it doesn't work ...: S Exactly what the url that I have sent has......

**Comment by joeyparrish** [comment]
That sounds like a coincidence to me.  jumpLargeGaps shouldn't have anything to do with it, and the content plays for me (with jumpLargeGaps and without) in ChromeOS.

**Comment by TheModMaker** [comment]
Your screenshot doesn't appear to show an error from us, just a console log.  If you click play, does the video play?  How about if you seek?  If you go to `about:config` (for Firefox), check that both `media.mp4.enabled` and `media.mediasource.mp4.enabled` configs are `true`.

**Comment by avelad** [comment]
If you click on play it doesn't work. if you do seek if it works. mp4 and mediasource are activated.

**Comment by TheModMaker** [comment]
So that log isn't relevant for some reason.  It comes from something else (maybe an extension?).  There is a small gap at the start of your content.  What may be happening is the browser sometimes doesn't play through it and we get stuck.  So it probably has to do with gap jumping not handling this correctly.  I don't see any logs about jumping the gap in your screenshot, so we're not jumping it correctly.

What are the values of `video.currentTime`, `video.buffered.length`, and `video.buffered.start(0)` when it is stalled?

**Comment by avelad** [comment]
![image](https://user-images.githubusercontent.com/8983024/109223363-c81e8d80-77ba-11eb-91e6-a9a53567f9a9.png)

```
document.getElementById('video').currentTime
0
document.getElementById('video').buffered.length
1
document.getElementById('video').buffered.start(0)
0.044
document.getElementById('video').buffered.end(0)
15.978666
```

**Comment by joeyparrish** [comment]
Sounds like https://github.com/google/shaka-player/commit/f7cc2f94312a970f356e9a1a8f47b9ab2216cef6 which landed after v3.0.8.  Is this fixed in the nightly build?

https://nightly-dot-shaka-player-demo.appspot.com/

**Comment by avelad** [comment]
You're right, it gets fixed with that commit. Thanks @joeyparrish
